doc/Makefile

```diff
@@ -47,13 +47,13 @@ api:
 
 
 linkcheck:
-	$(SPHINXBUILD) -b linkcheck $(ALLSPHINXOPTS) $(BUILDDIR)/linkcheck
```

Comment:
As mentioned in https://github.com/GenericMappingTools/pygmt/issues/914#issuecomment-780033253, we probably can remove the `linkcheck` target.

Reply:
Right, should we just remove linkcheck in this PR since the automated link checker CI is done at #634?

Reply:
OK to me.